extras/nginx/README.md

```diff
@@ -0,0 +1,154 @@
+Requirements
+------------
+
+Note that in addition to the dependencies required for [building quiche](https://github.com/cloudflare/quiche#building)
+you'll also need any dependency that might be required for [building NGINX](https://nginx.org/en/docs/configure.html)
+itself.
+
+NGINX will need to be built against BoringSSL, not OpenSSL, to work properly
+with quiche. This is done automatically during the build process described below.
+
+Once [OpenSSL merges support for QUIC](https://github.com/openssl/openssl/pull/8797)
+it will be possible to build using OpenSSL as well.
+
+Building
+--------
+
+The first step is to [download and unpack the NGINX source
+code](https://nginx.org/en/download.html). Note that the HTTP/3 and QUIC patch
+only works with the 1.16.x release branch (the latest stable release being
+1.16.1).
+
+```
+ % curl -O https://nginx.org/download/nginx-1.16.1.tar.gz
+ % tar xvf nginx-1.16.1.tar.gz
+```
+
+As well as quiche, the underlying implementation of HTTP/3 and QUIC:
+```
+ % git clone --recursive https://github.com/cloudflare/quiche
+```
+
+Next you’ll need to apply the patch to NGINX:
+```
+ % cd nginx-1.16.1
+ % patch -p01 < ../quiche/extras/nginx/nginx-1.16.patch
+```
+
+And finally build NGINX with HTTP/3 support enabled:
+```
+ % auto/configure                              \
+       --prefix=$PWD                           \
+       --with-http_ssl_module                  \
+       --with-http_v2_module                   \
+       --with-http_v3_module                   \
+       --with-openssl=../quiche/deps/boringssl \
+       --with-quiche=../quiche
+```
+
+The above command instructs the NGINX build system to enable the HTTP/3 support
+(`--with-http_v3_module`) by using the quiche library found in the path it was
+previously downloaded into (`--with-quiche=../quiche`).
+
+Running
+-------
+
+The following configuration file can be used as a starting point to enable
+HTTP/3 support:
+
+```
+events {
+    worker_connections  1024;
+}
+
+http {
+    server {
+        listen 0.0.0.0:443 ssl http2;
+        listen [::]:443 ssl http2;
+
+        listen 0.0.0.0:443 quic reuseport;
```

Comment:
Would it help to describe the `quic` directive in appropriate section of the readme?

Reply:
This mostly follows the NGINX HTTP/2 module documentation https://nginx.org/en/docs/http/ngx_http_v2_module.html which also doesn't do this. I think this is because the `listen` directive actually comes from a different module.

Reply:
that's fine by me

Reply:
I added comments to the example configuration to make this a bit more clear.